extensions/tensorflow/TFExtractTopLabels.cpp

```diff
@@ -134,9 +136,9 @@ int64_t TFExtractTopLabels::LabelsReadCallback::process(const std::shared_ptr<io
   buf.resize(buf_size);
 
   while (total_read < stream->size()) {
-    auto read = stream->read(reinterpret_cast<uint8_t *>(&buf[0]), static_cast<int>(buf_size));
-
-    for (auto i = 0; i < read; i++) {
+    const auto read = stream->read(reinterpret_cast<uint8_t *>(&buf[0]), buf_size);
+    if (io::isError(read)) break;
```

Comment:
not sure how other `Callback::process` methods work when they encounter an error, do they generally return what they managed to process or do they signal an error?

Reply:
read/write usually signal an error, but it's not consistent. I think Callback::process just forwards these results.